packages/create/src/cli/change/change-data-source.ts

```diff
@@ -0,0 +1,100 @@
+import type { Command } from 'commander';
+import type { AdpPreviewConfig, DescriptorVariant, PromptDefaults } from '@sap-ux/adp-tooling';
+import { generateChange, ChangeType, getPromptsForChangeDataSource, getManifest } from '@sap-ux/adp-tooling';
+import { getLogger, traceChanges } from '../../tracing';
+import { readFileSync, existsSync } from 'fs';
+import { join } from 'path';
+import { UI5Config } from '@sap-ux/ui5-config';
+import { promptYUIQuestions } from '../../common';
+
+let loginAttempts = 3;
+
+/**
+ * Add a new sub-command to change the data source of an adaptation project to the given command.
+ *
+ * @param {Command} cmd - The command to add the change data-source sub-command to.
+ */
+export function addChangeDataSourceCommand(cmd: Command): void {
+    cmd.command('data-source [path]')
+        .option('-s, --simulate', 'simulate only do not write or install')
+        .action(async (path, options) => {
+            await changeDataSource(path, { ...options }, !!options.simulate);
+        });
+}
+
+/**
+ * Changes the data source of an adaptation project.
+ *
+ * @param {string} basePath - The path to the adaptation project.
+ * @param {PromptDefaults} defaults - The default values for the prompts.
+ * @param {boolean} simulate - If set to true, then no files will be written to the filesystem.
+ */
+async function changeDataSource(basePath: string, defaults: PromptDefaults, simulate: boolean): Promise<void> {
+    const logger = getLogger();
+    try {
+        if (!basePath) {
+            basePath = process.cwd();
+        }
+        checkEnvironment(basePath);
+
+        const variant = getVariant(basePath);
+        const ui5Conf = await UI5Config.newInstance(readFileSync(join(basePath, 'ui5.yaml'), 'utf-8'));
+        const adp = ui5Conf.findCustomMiddleware<{ adp: AdpPreviewConfig }>('fiori-tools-preview')?.configuration?.adp;
```

Comment:
What do you mean? What is the issue here?

Reply:
If you generate the project with the open source middleware used directly and not wrapped in the `@sap/ux-ui5-tooling` (which is a valid usecase) then it doesn't work because your logic above cannot find it.

Reply:
Fixed with: fd93e76b7055e9d142e2d54416fe7dbf03fa2f1b